internal/devbox/devbox.go

```diff
@@ -1262,12 +1262,11 @@ func (d *Devbox) parseEnvAndExcludeSpecialCases(currentEnv []string) (map[string
 		// Finding nix executables in path and passing it through
 		// As well as adding devbox itself to PATH
 		// Both are needed for devbox commands inside pure shell to work
-		includedInPath, err := findNixInPATH(env)
+		nixPath, err := findNixInPATH()
```

Comment:
If `findNixInPATH` is only ever called with `os.Environ()`, then I think we should just remove it and simplify this down to:

```suggestion
		nixPath, err := exec.LookPath("nix")
		if err != nil {
				return "", errors.New("could not find any nix executable in PATH. Make sure Nix is installed and in PATH, then try again")
		}
		nixPath = filepath.Dir(nixPath)
```